src/main/java/Util/Storage.java

```diff
@@ -0,0 +1,183 @@
+                        if(strArr[1].equals("1")) task.doneJob();
+                        storedList.add(task);
+                    }
+                    //for Deadline
+                    if (strArr[i].equals("D")) {
+                        Deadline task = new Deadline(strArr[2], strArr[3]);
+                        if(strArr[1].equals("1")) task.doneJob();
+                        storedList.add(task);
+                    }
+                    //for Event
+                    if (strArr[i].equals("E")) {
+                        Event task = new Event(strArr[2], strArr[3]);
+                        if(strArr[1].equals("1")) task.doneJob();
+                        storedList.add(task);
+                    }
+                }
+            }
+        }
+        return storedList;
+    }
+
+    /**
+     * Closes off both the scanner and printwrite instances.
+     */
+    public void closeFile() {
+        scanner.close();
+        pw.close();
+    }
+
+    /**
+     * Deletes the file in filePath and rewrites the entire file through iterating
+     * the inputted TaskList
+     *
+     * @param taskList is the object for which the method will get data from
+     */
+
+    public void rewriteWriter(TaskList taskList) throws Exception{
+        String str = "";
+        for(int i = 0; i < taskList.getTaskList().size(); i++) {
+            Task task = taskList.getTaskList().get(i);
+            String taskType = taskList.getTaskList().get(i).toString().substring(1,2) +" | ";
+            String doneType = "";
+            if(taskList.getTaskList().get(i).toString().substring(4,5).equals("O")) doneType = "1 | ";
+            else doneType = "0 | ";
+            String after = taskList.getTaskList().get(i).toString().substring(7);
+            String desc = taskList.getTaskList().get(i).getDescription();
+            String atby = "";
+            if(taskList.getTaskList().get(i) instanceof Todo) {
+                atby = "";
+            } else if(taskList.getTaskList().get(i) instanceof Deadline) {
+                String[] strArr = after.split(" \\(by:");
+                atby = " |" + strArr[1].replace(")", "");
+            } else if(taskList.getTaskList().get(i) instanceof Event) {
+                String[] strArr = after.split( " \\(at:");
+                atby = " |" + strArr[1].replace(")", "");
+            } else {
+                throw new DukeException("task instance not declared yet");
+            }
+            if(i != 0) str += "\n";
+            str += taskType + doneType + desc + atby;
+
+        }
+        fw = new FileWriter(new File(this.filePath), false);
+        pw.print(str);
+        pw.flush();
+        fw.close();
+        fw = new FileWriter(new File(this.filePath), true);
+    }
+
+    /**
+     * Write a Todo task into the file from filePath
+     *
+     * @param task the Todo task which will be analysed
+     */
+    public void writeTodo(Task task) {
```

Comment:
You can actually combine all these into 1 function, or move them into the tasks themselves and extend them. That will reduce code duplication.